vllm/v1/engine/__init__.py

```diff
@@ -0,0 +1,72 @@
+import enum
+from dataclasses import dataclass
+from typing import List, Optional, Union
+
+import msgspec
+
+from vllm.lora.request import LoRARequest
+from vllm.sampling_params import RequestOutputKind, SamplingParams
+
+
+@dataclass
+class DetokenizerRequest:
+
+    request_id: str
+    prompt: Optional[str]
+    prompt_token_ids: List[int]
+    skip_special_tokens: bool
+    spaces_between_special_tokens: bool
+    output_kind: RequestOutputKind
+
+    stop: List[str]
+    include_stop_str_in_output: bool
+
+
+class EngineCoreRequest(msgspec.Struct, omit_defaults=True):
```

Comment:
Any reason to specify `array_like=True` and `gc=False` for the other cases but not this one? 

BTW did you see any performance improvement for `array_like=True`? We might want to make it easy to switch all of these over to `array_like=False` because
> This may improve performance, at the cost of a more inscrutable message encoding.

(from https://jcristharif.com/msgspec/api.html)

Not sure how inscrutable they get though

Reply:
Mostly that `EngineCoreRequests` are infrequent in comparison to `EngineCoreOutput`. 

So we wanted to have the error message encoding since it does not really matter for performance

Reply:
I measured some benefits from array like. Not monumental though. I think we can do a separate experiment as a follow up PR

Reply:
Where do we actually care about/log the wire-encoded contents though? This is for internal IPC so it shouldn't matter imo, better to be compact. It's like protobuf, I'm not sure how readable the msgpack encoding is anyhow.

That said it shouldn't be hard to make this a toggle as suggested.